crates/types/src/traits/election.rs

```diff
@@ -125,4 +125,32 @@ pub trait Membership<TYPES: NodeType>: Debug + Send + Sync {
 
     /// Returns the threshold required to upgrade the network protocol
     fn upgrade_threshold(&self, epoch: TYPES::Epoch) -> NonZeroU64;
+
+    #[allow(clippy::type_complexity)]
+    /// Handles notifications that a new epoch root has been created
+    /// Is called under a read lock to the Membership. Return a callback
+    /// with Some to have that callback invoked under a write lock.
+    fn add_epoch_root(
```

Comment:
Didn't we say we need this to be async?

Currently I imagine we would want to do external requests here or even if we de-couple it we may want to take out some async locks or read from channels.

Reply:
I think this can be non async but sync_l1 should be async?

Reply:
I don't really want to use `sync_l1` unless it can't be avoided. It shouldn't leak into consensus that we are using the L1 if we can avoid it.

Reply:
If we have only one function, then yes, it should be async due to calls to L1.